### PR TITLE
Support text-recognition-resnet-fc for text_detection_demo

### PR DESCRIPTION
**Pull request by akorobeinikov**
Add new `text-recognition-resnet-fc` model for text detection demo.

### CODE REVIEW

File: models/public/text-recognition-resnet-fc/model.yml [hunk 1]
```diff
@@ -291,7 +291,8 @@ conversion_to_onnx_args:
 model_optimizer_args:
   - --input_shape=[1,1,32,100]
   - --input=input
-  - --scale_values=input[255]
+  - --mean_values=input[0.5]
```
Comment (eaidova) [inline models/public/text-recognition-resnet-fc/model.yml:294]:
```suggestion
  - --mean_values=input[127.5]
```

Comment (eaidova) [reply models/public/text-recognition-resnet-fc/model.yml:294]: @morkovka1337 looks like your configuration in internal repo is incorrect

Comment (eaidova) [reply models/public/text-recognition-resnet-fc/model.yml:294]: @akorobeinikov could you please also reflect origina input/output and normalization parameters in model README.md

File: demos/text_detection_demo/cpp/models.lst [hunk 2]
```diff
@@ -1,4 +1,4 @@
 # This file can be used with the --list option of the model downloader.
 handwritten-score-recognition-????
 text-detection-????
-text-recognition-????
+text-recognition-*
```
Comment (eaidova) [inline demos/text_detection_demo/cpp/models.lst:4]: I think it may be useful to stay some hints regarding different models usage in model.lst

File: models/public/text-recognition-resnet-fc/model.yml (hunk 1 above)
Comment (morkovka1337) [reply models/public/text-recognition-resnet-fc/model.yml:294]: There is a preprocessing in the original model which could be confusing for the first time. The cause is that `Normalize` first divides input tensor on 255 and then normalizes with mean=0.5 and std=0.5.
https://albumentations.ai/docs/api_reference/augmentations/transforms/#albumentations.augmentations.transforms.Normalize
It should be double checked but it seems to me that simplified equation is:
(x / 255 - 0.5) / 0.5 = x / 127.5 - **1**
Thus, preprocessing should be: 
```
--scale_values=input[127.5]
--mean_values=input[1]
```

But again, this should be double checked.

Comment (eaidova) [reply models/public/text-recognition-resnet-fc/model.yml:294]: MO and AC use another normalization ops  order
(x - mean) / std

If you checked with previous version of AC config and model reproduces accuracy, then (x / 255 - 0.5) / 0.5 = (x - 127.5) / 127.5

Comment (morkovka1337) [reply models/public/text-recognition-resnet-fc/model.yml:294]: Ok then, your version should be correct. I did not check it, it was just theoretical calculation. =)

File: demos/text_detection_demo/cpp/models.lst (hunk 2 above)
Comment (akorobeinikov) [reply demos/text_detection_demo/cpp/models.lst]: Changed normalization, change ac metrics and add info about normalization to README.md. Add information about `decoder_type` parameter to models.lst.

**Review by eaidova — APPROVED** [review]

File: demos/text_detection_demo/cpp/text_detection_demo.hpp [hunk 3]
```diff
@@ -55,6 +56,7 @@ static const char decoder_bandwidth_message[] = "Optional. Bandwidth for CTC bea
 DEFINE_bool(h, false, help_message);
 DEFINE_string(m_td, "", text_detection_model_message);
 DEFINE_string(m_tr, "", text_recognition_model_message);
+DEFINE_string(dt, "", text_recognition_decoder_type_message);
```
Comment (morkovka1337) [inline demos/text_detection_demo/cpp/text_detection_demo.hpp:59]:
```suggestion
DEFINE_string(dt, "ctc", text_recognition_decoder_type_message);
```

If provide default option for decoder, old text recognition models (12 & 13) could be used as is (without setting additional parameters). 
Also, if not provide default choice, documentation for text-recognition-0012 should also be changed.

File: demos/text_detection_demo/cpp/README.md [hunk 4]
```diff
@@ -64,10 +66,13 @@ For example, use the following command line command to run the application:
 ./text_detection_demo -m_td <path_to_model>/text-detection-0004.xml \
                       -m_tr <path_to_model>/text-recognition-0013.xml \
                       -i <path_to_image>/sample.jpg \
+                      -dt "ctc" \
                       -tr_pt_first \
                       -tr_o_blb_nm "logits"
 ```
 
+For `text-recognition-resnet-fc` you should use `simple` decoder for `-dt` option. For other models `ctc` decoder.
```
Comment (morkovka1337) [inline demos/text_detection_demo/cpp/README.md:74]:
```suggestion
For `text-recognition-resnet-fc` you should use `simple` decoder for `-dt` option. For other models use `ctc` decoder.
```

File: demos/text_detection_demo/cpp/README.md [hunk 5]
```diff
@@ -34,7 +34,7 @@ Options:
     -limit "<num>"             Optional. Number of frames to store in output. If 0 is set, all frames are stored.
     -m_td "<path>"               Required. Path to the Text Detection model (.xml) file.
     -m_tr "<path>"               Required. Path to the Text Recognition model (.xml) file.
-    -dt "<path>"               Required. Type of the decoder, either 'simple' for SimpleDecoder or 'ctc' for CTC greedy and CTC beam search decoders.
+    -dt "<type>"               Required. Type of the decoder, either 'simple' for SimpleDecoder or 'ctc' for CTC greedy and CTC beam search decoders.
```
Comment (eaidova) [inline demos/text_detection_demo/cpp/README.md:37]: please update help message. You define default value, so it is not required anymore

File: demos/text_detection_demo/cpp/text_detection_demo.hpp [hunk 6]
```diff
@@ -17,6 +17,7 @@ DEFINE_OUTPUT_FLAGS
 static const char help_message[] = "Print a usage message.";
 static const char text_detection_model_message[] = "Required. Path to the Text Detection model (.xml) file.";
 static const char text_recognition_model_message[] = "Required. Path to the Text Recognition model (.xml) file.";
+static const char text_recognition_decoder_type_message[] = "Optional. Type of the decoder, either 'simple' for SimpleDecoder or 'ctc' for CTC greedy and CTC beam search decoders.";
```
Comment (eaidova) [inline demos/text_detection_demo/cpp/text_detection_demo.hpp:20]:
```suggestion
static const char text_recognition_decoder_type_message[] = "Optional. Type of the decoder, either 'simple' for SimpleDecoder or 'ctc' for CTC greedy and CTC beam search decoders. Default is 'ctc'";
```

**Review by morkovka1337 — APPROVED** [review]

**Comment by eaidova** [comment]
Jenkins please retry a build
